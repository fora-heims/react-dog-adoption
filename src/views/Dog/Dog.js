import { React, useState, useEffect } from 'react';
import { useParams, useHistory } from 'react-router-dom';
import DogDetail from '../../components/DogDetail/DogDetail.js';
import Header from '../../components/Header/Header.js';
import { getDog, deleteDog } from '../../services/dogs.js';

export default function Dog() {
  const [dog, setDog] = useState({});
  const [loading, setLoading] = useState(true);

  const params = useParams();
  const history = useHistory();

  useEffect(() => {
    const fetchData = async () => {
      const doggyData = await getDog(params.id);
      setDog(doggyData);
      setLoading(false);
    };
    fetchData();
  }, [params.id]);

  const editButtonHandler = () => {
    history.push(`/dogs/${params.id}/edit`);
  };

  const deleteButtonHandler = async () => {
    await deleteDog(params.id);
    history.push('/');
  };

  return (
    <div>
      <Header />
      {loading ? (
        <span>...loading</span>
      ) : (
        <DogDetail
          deleteButtonHandler={deleteButtonHandler}
          dog={dog}
          editButtonHandler={editButtonHandler}
          showDetail
        />
      )}
    </div>
  );
}
