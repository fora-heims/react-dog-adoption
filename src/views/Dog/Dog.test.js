import Dog from './Dog.js';
import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';

test('', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/dogs/17']}>
      <Route path="/dogs/:id">
        <Dog />
      </Route>
    </MemoryRouter>
  );
  await screen.findByText('Meet Mrs. Weasley');
  expect(container).toMatchSnapshot();
});
