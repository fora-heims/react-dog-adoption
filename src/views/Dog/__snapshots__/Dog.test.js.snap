// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div>
  <div>
    <div
      class="detail"
    >
      <h3>
        Meet 
        Mrs. Weasley
      </h3>
      <img
        alt="dog"
        class="big-dog"
        src="https://placedog.net/1000/563?id=3"
      />
      <p>
        Mrs. Weasley
         is a(n) 
        3
         year old 
        Golden Retriever
      </p>
      <br />
      <span>
        Mrs. Weasley was found pregnant on the streets. She gave birth to 4 lively pups, Ron & Ginny are still available for adoption. She's working on her resource guarding but will make a wonderful family dog. 
      </span>
    </div>
  </div>
</div>
`;
